frontend/src/routing/associations.ts

```diff
@@ -14,85 +14,109 @@ import { AssociationMarketplaceProductAdministration } from "../components/assoc
 import { AssociationMarketplaceOrders } from "../components/associations/marketplace/Orders";
 import { AssociationMarketplaceCounter } from "../components/associations/marketplace/Counter";
 
+/**
+ * Association routes are like common routes, with  one additional parameter,
+ * which is `defaultLayout`. If `defaultLayout` is `true`, the `component` will
+ * be automatically be wrapped in an `AssociationLayout`, which includes a sidebar
+ * by default. If it's `false` it will not.
+ * You may want to take advantage of it if :
+ * - you want a custom sidebar, in that case you'll wrap your component with
+ * `AssociationLayout` manually
+ * - you don't want a sidebar at all, in that case you can organise your code as
+ * usual
+ */
 export const routes = (association) => [
```

Comment:
Would it be possible to type `routes`?
I guess something like
`(association: Association) => {path: string; component: any; exact: boolean; props: {[key:string]: any}; defaultLayout?: boolean}`